File: src/main/java/com/lowtuna/jsonblob/view/EditorView.java
package com.lowtuna.jsonblob.view;

import com.lowtuna.dropwizard.extras.config.GoogleAnalyticsConfig;
import lombok.Data;

import java.util.Set;

@Data
public class EditorView extends JsonBlobView {
    private String blobId;
    private String jsonBlob;

    public EditorView(String gaWebPropertyID, String pageName, Set<GoogleAnalyticsConfig.CustomTrackingCode> customTrackingCodes) {
        super("/editor.hbs", gaWebPropertyID, pageName, customTrackingCodes);
    }
}

File: src/main/kotlin/jsonblob/api/http/view/EditorView.kt
package jsonblob.api.http.view

import io.micronaut.core.annotation.Introspected
import jsonblob.config.GoogleAnalyticsConfig

@Introspected
class EditorView(
    gaWebPropertyID: String,
    pageName: String,
    gaCustomTrackingCodes: Set<GoogleAnalyticsConfig.CustomTrackingCode>,
    val jsonBlob: String? = null,
    val blobId: String? = null
) : JsonBlobView(gaWebPropertyID, pageName, gaCustomTrackingCodes)